### PR TITLE
Custom esports images for teams

### PR DESCRIPTION
**Pull request by AndrewJakubowicz**
# Context

For certain esports teams we want to be able to add custom images. This PR adds an internal admin Clan edit page for editing clans and uploading custom images.

New internal admin tool at route: `/admin/clan/:clanID`

The new `esportsImage` property uploads into our static file system and replaces the static image on the teams league page.

These gifs provide more context.

Uploading a new image:
dbe2e0f84c586da159e1bc81252b2c86

Saving the uploaded image:
9077b3102c5010d6f89c24c38c049bae

The final result: 
image

Vs the default static image:
image

# Testing

As this PR includes mostly UI additions, it was manually tested against a local mongo instance.
To successfully reproduce the gif animations above, the server portion needs to be running locally - this allows admin users to edit the esportsImage and clan in the put request.

# Risk

Low risk as most of the logic is internal tooling that doesn't need to be perfect. Few changes to user facing code.
The team/clan admin UI is secured server side.

### CODE REVIEW

File: app/views/admin/PageClanEdit.vue [hunk 1]
```diff
@@ -0,0 +1,95 @@
+<script>
+import { getClan } from 'core/api/clans'
+import CocoCollection from 'app/collections/CocoCollection'
+import Clan from 'app/models/Clan'
+
+const api = require('core/api')
+require('core/services/filepicker')()
```
Comment (nwinter) [inline app/views/admin/PageClanEdit.vue:7]: Is this needed here, or is it taken care of [when an ImageFileTreema is created](https://github.com/codecombat/ozaria/blob/4478a19c21224f0053f6bca4458635bbf2549648/app/core/treema-ext.coffee#L238) anyway?

File: app/views/admin/PageClanEdit.vue [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    <div v-if="clan !== null">
+      <p>This internal tool lets admin and internal accounts customize teams.
+        For engineers - 'team' refers to a 'clan' in the code and database.
+        Use the tool for:
+      </p>
+      <ul>
+        <li>censoring a name or description on a team</li>
+        <li>changing or adding an owner user to a team</li>
+        <li>adding a custom image for a team</li>
+        <li>editting any other data of a team</li>
```
Comment (nwinter) [inline app/views/admin/PageClanEdit.vue:78]:
```suggestion
        <li>editing any other data of a team</li>
```

File: app/views/landing-pages/league/components/ClanSelector.vue [hunk 3]
```diff
@@ -12,6 +12,12 @@ export default {
       required: false,
       default: ''
     }
+  },
+
+  computed: {
+    clanSanitized () {
```
Comment (nwinter) [inline app/views/landing-pages/league/components/ClanSelector.vue:18]:
```suggestion
    clansSanitized () {
```

File: app/views/landing-pages/league/components/ClanSelector.vue [hunk 4]
```diff
@@ -21,7 +27,7 @@ export default {
     <label for="clans">My Teams:</label>
     <select id="clans" name="clans" @change="e => $emit('change', e)">
       <option value="global" :selected="selected===''">--</option>
-      <option  v-for="clan in clans" :key="clan._id" :value="clan._id" :selected="selected===clan._id">
+      <option  v-for="clan in clanSanitized" :key="clan._id" :value="clan._id" :selected="selected===clan._id">
```
Comment (nwinter) [inline app/views/landing-pages/league/components/ClanSelector.vue:30]:
```suggestion
      <option  v-for="clan in clansSanitized" :key="clan._id" :value="clan._id" :selected="selected===clan._id">
```

**Review by nwinter — APPROVED** [review]
Looks great! Picked a few nits.

File: app/views/admin/PageClanEdit.vue (hunk 1 above)
Comment (AndrewJakubowicz) [reply app/views/admin/PageClanEdit.vue:7]: Excellent catch. Will remove and double check functionality.

I can think of some other places then we are double initializing in internal tools.
Thanks!

Comment (AndrewJakubowicz) [reply app/views/admin/PageClanEdit.vue]: Tested and can be removed

**Pull request merged**